Adaptation Bibliotheque nationale de France.js for the new BnF catalog

facem opened the pull request:
The changes consist on :
the Import works whatever display style (intermarc, unimarc, public)
in the field 'URL' there is a ARK link of the notice
the URL of numeric document is in field "attachement document"
correction for the role detection
The document type detection uses the field Unimarc
Modification of the field "source"

adam3smith commented on Bibliotheque nationale de France.js:
```diff
@@ -41,7 +41,8 @@ var BnfClass = function() {
 	*/
 	function getCreatorType(aut) {
-		switch(aut['4']) {
+		typeAut = aut['4'].replace(/\s/g,"");
```
should be `var typeAut` and I'd use `.trim()` instead of the replace.

adam3smith commented on Bibliotheque nationale de France.js:
```diff
@@ -499,9 +469,11 @@ var BnfClass = function() {
 		try {
-			var xPath = '//div[@class="notice-detail"]//div[@id="type"]';
+			// var xPath = '//div[@class="notice-detail"]//div[@id="type"]';
+			var xPath= './/*[@id="corps"]/div[1]/div[3]/div/h1';
```
this seems very fragile. No way to identify this with an attribute or leave out the `div[1]/div[3]`?

adam3smith commented on Bibliotheque nationale de France.js:
```diff
@@ -510,15 +482,18 @@ var BnfClass = function() {
 			return "book";
-		case "sound recording":
-			return "audioRecording";
+		case "Notice de périodique":
```
this seems wrong, no? shouldn't this be `newspaperArticle`?

adam3smith commented on Bibliotheque nationale de France.js:
```diff
@@ -499,9 +469,11 @@ var BnfClass = function() {
 	*/
 	this.getDCType = function(doc, url) {
```
since these don't appear to be DCTypes anymore, rename this function

adam3smith commented on Bibliotheque nationale de France.js:
```diff
@@ -510,15 +482,18 @@ var BnfClass = function() {
 
 	/* Translate a DC type to a corresponding Zotero item type. Currently obsolete. */
 	this.translateDCType = function(type) {
```
rename this one, too, and change the comment.

adam3smith commented:
@symac -- if you have a minute to look over this? Makes a lot of sense to me, but you wrote most of the code.

facem commented:
Hello,
I have corrected.
I have deleted the method that didn't work with all display.
I hadn't made before because I did not want to change too much the code

facem commented:
Hello,
I have corrected and simplidied the code.
I have deleted the method that didn't work with all display.
I hadn't made before because I did not want to change too much the code

Best regard

2016-04-22 16:41 GMT+02:00 Sebastian Karcher <notifications@github.com>:

> @symac <https://github.com/symac> -- if you have a minute to look over
> this? Makes a lot of sense to me, but you wrote most of the code.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/zotero/translators/pull/1051#issuecomment-213453398>
>

adam3smith commented on Bibliotheque nationale de France.js:
```diff
@@ -41,7 +41,9 @@ var BnfClass = function() {
 	function getCreatorType(aut) {
-		switch(aut['4']) {
+		//typeAut = aut['4'].replace(/\s/g,"");
+		typeAut = aut['4'].trim();
```
you're still not declaring the variable here. Needs to be `var typeAut = ...`

adam3smith commented:
Looks good to me otherwise.

facem commented:
Hello 
I'm sorry for this mistake.
I have just corrected.
Thank you for the time that you spend for me and for your help
Best regard

symac commented:
Trying the new translator I have encountered an error when trying to get list (but I believe they also exist in the previous version, seems BnF has updated their markup). In getSelectedItems function, there is the following line : 
`var link = doc.evaluate('./div[@class="notice-contenu"]/a', cell, null, XPathResult.ANY_TYPE, null).iterateNext();`

that needs to be replaced by : 
`var link = doc.evaluate('./div[@class="notice-contenu"]/div[@class="notice-synthese"]/a', cell, null, XPathResult.ANY_TYPE, null).iterateNext();`

The div with the class notice-synthese has been added meanwhile I believe. Otherwise I have not found any error with some tests, seems ok to me.

facem commented:
Hello
Indeed, I have just corrected and I have change the  code for don't do nothing when the page  type is "undefined".

Best regard

facem commented:
Hello,

I corrected the last error.

Is it good?

what happens when it is ok ?

Best regard

Le 9 mai 2016 14:35, "Sylvain Machefert" <notifications@github.com> a
écrit :
>
> Trying the new translator I have encountered an error when trying to get
list (but I believe they also exist in the previous version, seems BnF has
updated their markup). In getSelectedItems function, there is the following
line :
> var link = doc.evaluate('./div[@class="notice-contenu"]/a', cell, null,
XPathResult.ANY_TYPE, null).iterateNext();
>
> that needs to be replaced by :
> var link =
doc.evaluate('./div[@class="notice-contenu"]/div[@class="notice-synthese"]/a',
cell, null, XPathResult.ANY_TYPE, null).iterateNext();
>
> The div with the class notice-synthese has been added meanwhile I
believe. Otherwise I have not found any error with some tests, seems ok to
me.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
Le 9 mai 2016 14:35, "Sylvain Machefert" <notifications@github.com> a
écrit :
>
> Trying the new translator I have encountered an error when trying to get
list (but I believe they also exist in the previous version, seems BnF has
updated their markup). In getSelectedItems function, there is the following
line :
> var link = doc.evaluate('./div[@class="notice-contenu"]/a', cell, null,
XPathResult.ANY_TYPE, null).iterateNext();
>
> that needs to be replaced by :
> var link =
doc.evaluate('./div[@class="notice-contenu"]/div[@class="notice-synthese"]/a',
cell, null, XPathResult.ANY_TYPE, null).iterateNext();
>
> The div with the class notice-synthese has been added meanwhile I
believe. Otherwise I have not found any error with some tests, seems ok to
me.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub

adam3smith commented:
thanks!

facem commented:
Hello
Great
Thanks for all.
Best reagard

2016-05-19 15:26 GMT+02:00 Sebastian Karcher <notifications@github.com>:

> thanks!
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/zotero/translators/pull/1051#issuecomment-220323409>
>
